src/lib/tachyon-utils.js

```diff
@@ -66,11 +71,139 @@ const getEDLDevice = async ({ ui = new UI() } = {}) => {
 		}
 		await delay(DEVICE_READY_WAIT_TIME);
 	}
-};
+}
+
+async function prepareFlashFiles({ logFile, ui, partitionsList, dir = process.cwd(), deviceId, operation, checkFiles = false } = {}) {
```

Comment:
I was thinking about transforming this into a class named `PartitionManager` or something like that but I started with just one orchestrator function here.